File: backend/src/performance/002-replace_substring/rs.cpp
#include <iostream>
#include <string>
#include <stdlib.h>

using namespace std;

void rs(long n);

main(int argc, char **argv) {
    long n = strtol(argv[1], NULL, 10);

    rs(n);

    return 0;
}

void rs(long n) {
    for (int i=0; i<n; i++) {
        string str = "Salida de emergencia";
        string substitution = "Entrada";
        string result = str.replace(str.begin(), str.begin()+6, substitution);
        //cout << result << endl;
    }
}

File: backend/src/performance/002-replace_substring/rs.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

from sys import argv

def rs():
    for _ in range(int(argv[1])):
        string = "Salida de emergencia"
        pattern = "Salida"
        substitution = "Entrada"
        string = string.replace(pattern, substitution)

if __name__ == "__main__":
    rs()